WordPress/src/main/java/org/wordpress/android/ui/media/MediaAddFragment.java

```diff
@@ -224,7 +243,16 @@ private void queueFileForUpload(String path) {
     }
 
     private void startMediaUploadService() {
-        getActivity().startService(new Intent(getActivity(), MediaUploadService.class));
+        if (NetworkUtils.isNetworkAvailable(getActivity())) {
+            getActivity().startService(new Intent(getActivity(), MediaUploadService.class));
+        }
+    }
+
+    private void resumeMediaUploadService() {
```

Comment:
Hate to nitpick too much but I notice this method duplicates onResume. Could you call it there? Should be ready to merge after that.